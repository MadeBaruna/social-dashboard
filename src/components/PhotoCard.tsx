import React from 'react';
import { Card } from 'semantic-ui-react';
import ImageLoader from './ImageLoader';

interface IProps {
  thumbnailUrl: string;
  openImage: () => void;
}

const PhotoCard = ({ openImage, thumbnailUrl }: IProps) => (
  <Card onClick={openImage} fluid>
    <ImageLoader imageUrl={thumbnailUrl} />
  </Card>
);

export default PhotoCard;
