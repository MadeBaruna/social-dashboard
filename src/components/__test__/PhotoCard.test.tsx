import React from 'react';
import { shallow } from 'enzyme';
import PhotoCard from '../PhotoCard';

it('renders PhotoCard components', () => {
  const wrapper = shallow(
    <PhotoCard thumbnailUrl="http://localhost/img.jpg" openImage={jest.fn()} />,
  );
  expect(wrapper).toMatchSnapshot();
});
