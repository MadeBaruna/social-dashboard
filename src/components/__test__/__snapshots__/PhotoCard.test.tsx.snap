// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PhotoCard components 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PhotoCard
    openImage={[MockFunction]}
    thumbnailUrl="http://localhost/img.jpg"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ImageLoader
        imageUrl="http://localhost/img.jpg"
      />,
      "fluid": true,
      "onClick": [MockFunction],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "imageUrl": "http://localhost/img.jpg",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ImageLoader
          imageUrl="http://localhost/img.jpg"
        />,
        "fluid": true,
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "imageUrl": "http://localhost/img.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
